collated/test/A0135812L.md

```diff
@@ -0,0 +1,445 @@
+        assertIndexNotFoundBehaviorForCommand("select");
+    }
+
+    @Test
+    public void execute_select_jumpsToCorrectTask() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        List<Task> threeTasks = helper.generateTaskList(3);
+
+        ToDo expectedAB = helper.generateToDo(threeTasks);
+        helper.addToModel(model, threeTasks);
+
+        assertCommandBehavior("select 2",
+                String.format(SelectCommand.MESSAGE_SELECT_TASK_SUCCESS, 2),
+                expectedAB,
+                expectedAB.getTaskList());
+        assertEquals(1, targetedJumpIndex);
+        assertEquals(model.getFilteredTaskList().get(1), threeTasks.get(1));
+    }
+
+
+    @Test
+    public void execute_deleteInvalidArgsFormat_errorMessageShown() throws Exception {
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE);
+        assertIncorrectIndexFormatBehaviorForCommand("delete", expectedMessage);
+    }
+
+    @Test
+    public void execute_deleteIndexNotFound_errorMessageShown() throws Exception {
+        assertIndexNotFoundBehaviorForCommand("delete");
+    }
+
+    @Test
+    public void execute_delete_removesCorrectTask() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        List<Task> threeTasks = helper.generateTaskList(3);
+
+        ToDo expectedAB = helper.generateToDo(threeTasks);
+        expectedAB.removeTask(threeTasks.get(1));
+        helper.addToModel(model, threeTasks);
+
+        assertCommandBehavior("delete 2",
+                String.format(DeleteCommand.MESSAGE_DELETE_TASK_SUCCESS, threeTasks.get(1)),
+                expectedAB,
+                expectedAB.getTaskList());
+    }
+
+
+    @Test
+    public void execute_find_invalidArgsFormat() throws Exception {
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE);
+        assertCommandBehavior("find ", expectedMessage);
+    }
+
+    @Test
+    public void execute_find_matchesSubstringInNames() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        Task pTarget1 = helper.generateTaskWithName("bla bla KEY bla");
+        Task pTarget2 = helper.generateTaskWithName("blaKEYbla bceofeia");
+        Task p1 = helper.generateTaskWithName("KE Y");
+        Task p2 = helper.generateTaskWithName("KE YKE YKE Ysduauo");
+
+        List<Task> fourTasks = helper.generateTaskList(p1, pTarget1, p2, pTarget2);
+        ToDo expectedAB = helper.generateToDo(fourTasks);
+        List<Task> expectedList = helper.generateTaskList(pTarget1, pTarget2);
+        helper.addToModel(model, fourTasks);
+
+        assertCommandBehavior("find KEY",
+                Command.getMessageForTaskListShownSummary(expectedList.size()),
+                expectedAB,
+                expectedList);
+    }
+
+    @Test
+    public void execute_find_isNotCaseSensitive() throws Exception {
```

Comment:
Refer naming convention for test methods again. The name should have no more than 3 parts.